### ISSUE TITLE
coda_read_definitions_fuzzer: Use-of-uninitialized-value in strhash

### ISSUE THREAD

**Issue by schwehr** [stcorp/coda#69]
```
==234470==WARNING: MemorySanitizer: use-of-uninitialized-value
    #0 0x55c6cc6f38ae in strhash third_party/stcorp_coda/libcoda/hashtable.c:99:5
    #1 0x55c6cc6f1f0b in coda_hashtable_insert_name third_party/stcorp_coda/libcoda/hashtable.c:155:37
    #2 0x55c6cc6f3b70 in coda_hashtable_add_name third_party/stcorp_coda/libcoda/hashtable.c:294:12
    #3 0x55c6cc6f684b in get_entries third_party/stcorp_coda/libcoda/ziparchive.c:287:13
    #4 0x55c6cc6f56a9 in coda_za_open third_party/stcorp_coda/libcoda/ziparchive.c:392:9
    #5 0x55c6cc5b9560 in read_definition_file third_party/stcorp_coda/libcoda/coda-definition-parse.c:4247:10
    #6 0x55c6cc5b9276 in coda_read_definitions third_party/stcorp_coda/libcoda/coda-definition-parse.c:4461:21
    #7 0x55c6cc5744cd in LLVMFuzzerTestOneInput third_party/stcorp_coda/fuzz/coda_read_definitions_fuzzer.cc:19:3
```

```c
if (read(zf->fd, entry->filename, entry->filename_length) < 0) // <--- allows short read to go forward
{
    zf->handle_error(strerror(errno));
    return -1;
}
entry->filename[entry->filename_length] = '\0';

if (hashtable_add_name(zf->hash_data, entry->filename) != 0)  // <-- msan in here
{
    zf->handle_error("zip file '%s' contains two entries with the same name '%s'", zf->filename,
                     entry->filename);
    return -1;
}
```

coda_read_definitions_fuzzer.c is currently very limited:

```c
extern "C" {
int coda_read_definitions(const char *definition_path);
}

extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
  FuzzerTemporaryFile temp_file(data, size);

  coda_init();
  auto done = absl::MakeCleanup([] { coda_done(); });

  coda_read_definitions(temp_file.filename());

  return 0;
}
```

[testcase-5144684351717376.zip](https://github.com/stcorp/coda/files/4607634/testcase-5144684351717376.zip)

**Comment by svniemeijer** [comment]
Additional checks implemented in b65b59e314470cc18ce3521f577a243507e172b9

**Comment by schwehr** [comment]
Verified
